cpp/src/arrow/buffer-test.cc

```diff
@@ -141,6 +141,19 @@ TEST(TestMutableBuffer, Wrap) {
   ASSERT_EQ(4, values[1]);
 }
 
+TEST(TestSTLStringBuffer, STLStringBuffer) {
+  std::string expected = "input data";
+
+  std::shared_ptr<STLStringBuffer> buffer;
+  {
+    std::string data_str = "input data";
+    ASSERT_OK(AllocateSTLStringBuffer(data_str, &buffer));
+  }
+
+  ASSERT_EQ(0, memcmp(buffer->data(), expected.c_str(), expected.size()));
+  ASSERT_EQ(static_cast<int64_t>(expected.size()), buffer->size());
```

Comment:
You should test that `expected` is empty now (not sure what are the expected semantics for a `std::string` after it has been moved

Reply:
``std::move`` does not define a specific state for the state of the moved memory, just cautions that the original reference should not be treated as valid anymore